### PR TITLE
[MNE-27, MKC-1233] ESP-NOW Tutorial

### PR DESCRIPTION
**Pull request by Hannes7eicher**
## What This PR Changes
This PR a tutorial about ESP-NOW

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+### Low Latency
+As a result of the limited data quantities and its lightweight design ESP-NOW introduces a low latency, which means that devices can quickly exchange data making it suitable for remote controlling other devices, e.g. a radio car.
+
+## Limitations
+
+### Limited Range 
+ESP-NOW has a limited signal range, typically around 220 meters under ideal conditions. The actual range can vary depending on factors like environmental interference, antenna design, and obstacles in the communication path.
+
+### Interference 
+Like other wireless communication technologies operating in the 2.4GHz band, ESP-NOW can be susceptible to interference from other devices and Wi-Fi networks. It's essential to choose communication channels carefully to minimize interference and ensure reliable communication.
```
Comment (kateryna-lukina) [inline content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md:62]:
```suggestion
Like other wireless communication technologies operating in the 2.4 GHz band, ESP-NOW can be susceptible to interference from other devices and Wi-Fi networks. It's essential to choose communication channels carefully to minimize interference and ensure reliable communication.
```

File: content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+### No Network Infrastructure
+ESP-NOW is designed for point-to-point and point-to-multipoint communication but doesn't provide the infrastructure for building complex network topologies. If your application requires a network with multiple interconnected devices or internet connectivity, you may need to complement ESP-NOW with additional networking solutions.
```
Comment (kateryna-lukina) [inline content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md:65]:
```suggestion
ESP-NOW is designed for point-to-point and point-to-multipoint communication but doesn't provide the infrastructure for building complex network topologies. If your application requires a network with multiple interconnected devices or Internet connectivity, you may need to complement ESP-NOW with additional networking solutions.
```

File: content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+ESP-NOW does not provide built-in acknowledgment mechanisms for confirming successful message delivery. If message reliability is crucial for your application, you'll need to implement your own acknowledgment and error-handling mechanisms.
+
+### Compatibility
+ESP-NOW is primarily designed for use with ESP8266 and ESP32 microcontrollers. While it can be compatible with other ESP32-based devices, it may not work seamlessly with other non-ESP platforms. Compatibility should be considered when designing a system with multiple types of devices.
+
+## Arduino IoT Cloud
+
+Arduino provides their own service for handling wireless communication called Arduino IoT Cloud. You can read more about it [here](https://docs.arduino.cc/arduino-cloud/).
+
+While the Arduino IoT Cloud offers similar features to the ESP-NOW protocols they are vastly different and should be used for specific usecases. Check the comparison table below for more information.
```
Comment (kateryna-lukina) [inline content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md:83]:
```suggestion
While the Arduino IoT Cloud offers similar features to the ESP-NOW protocols they are vastly different and should be used for specific use cases. Check the comparison table below for more information.
```

File: content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+| Data Size           | Unlimited data (min. Entry plan)                                            | max 250 bytes.                                                                 |
+| Device Compatibility| Compatible with various Arduino boards and IoT devices.                     | Primarily used with ESP8266 and ESP32 microcontrollers.                        |
+| Protocol            | Uses MQTT for communication.                                                | Uses a custom ESP-NOW protocol.                                                |
+| Power Consumption   | May consume more power, especially when connected to the internet.          | Known for low power consumption.                                               |
+| Data Processing     | Allows for cloud-based data processing and analytics.                       | Primarily for direct device-to-device communication.                           |
+| Use Cases           | IoT projects requiring global connectivity and cloud-based data management. | Applications needing low-latency, local communication.                         |
```
Comment (kateryna-lukina) [inline content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md:95]:
```suggestion
|                     | Arduino IoT Cloud                                                           | ESP-NOW                                                                        |
|---------------------|-----------------------------------------------------------------------------|--------------------------------------------------------------------------------|
| Range               | Depends on Internet connectivity, suitable for global reach.                | Approx. 220 m.                                                                   |
| Security            | Secure communication with encryption and authentication.                    | Basic security, may need additional measures.                                  |
| Delay               | Network-related delays due to Internet communication.                       | Low latency for local communication.                                           |
| Data Size           | Unlimited data (min. Entry plan)                                            | max 250 bytes.                                                                 |
| Device Compatibility| Compatible with various Arduino boards and IoT devices.                     | Primarily used with ESP8266 and ESP32 microcontrollers.                        |
| Protocol            | Uses MQTT for communication.                                                | Uses a custom ESP-NOW protocol.                                                |
| Power Consumption   | May consume more power, especially when connected to the internet.          | Known for low power consumption.                                               |
| Data Processing     | Allows for cloud-based data processing and analytics.                       | Primarily for direct device-to-device communication.                           |
| Use Cases           | IoT projects requiring global connectivity and cloud-based data management. | Applications needing low-latency, local communication.                         |
```

**Review by kateryna-lukina — COMMENTED** [review]
Some minor fixes, otherwise lgtm 😍

File: content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
+---
+title: ESP-NOW
```
Comment (karlsoderby) [inline content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md:2]:
```suggestion
title: Device to Device Communication with ESP-NOW
```

File: content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+description: ESP-NOW is a short distance wireless communication protocol develop by Espressive.
```
Comment (karlsoderby) [inline content/hardware/03.nano/boards/nano-esp32/tutorials/esp-now/esp-now.md:3]:
```suggestion
description: Learn how to use the ESP-NOW protocol to send data between ESP based boards.
```

**Review by karlsoderby — APPROVED** [review]

**Pull request merged**